S3Proxy/cache.go

```diff
@@ -5,40 +5,62 @@ import (
 	"time"
 )
 
-var s3Buckets = map[string]*s3BucketCacheItem{}
+var s3Buckets = map[string]*S3BucketCacheItem{}
+var s3Objects = map[string]*S3ObjectCacheItem{}
 
-type s3BucketCacheItem struct {
+// S3BucketCacheItem is a cache entry representing bucket info
+type S3BucketCacheItem struct {
```

Comment:
Not completed PR yet, but for the different cache items think about refactoring to use a generic CacheItem. For example:

```go
type CacheItem struct {
	Timestamp time.Time
	TTL       time.duration
}

func (c CacheItem) TimeLeft time.duration {
	tmp := &c
	tmp.TTL = o.TTL - time.Since(c.Timestamp)
	return tmp.TTL
}

type S3BucketCacheItem struct {
	CacheItem
	Name     string
	Location string
}
```